text/0000-int-guidelines.md

```diff
@@ -0,0 +1,164 @@
+
+If you are passing a value provided by one API into another API, is it
+generally safe to cast a **smaller** sized value into a **larger** sized
+value.
+
+For example, it is generally safe to cast an `i32` to an `i64`. It is also
+generally safe to case a `usize` to a `u64`, since it will never be bigger
+than `u64`.
+
+However, you should avoid **truncating** casts, which cast a larger sized
+integer to a smaller sized integer. You should also avoid casting between
+signed and unsigned integers (in either direction).
+
+If you need to truncate a number, or cast between signed and unsigned
+types, you should carefully consider what will happen if the source number
+is outside the bounds of the target type, and handle that case explicitly.
+
+# Drawbacks
+
+The primary drawback of this approach is the same as the drawback of
+not having a default integer type in the first place: unnecessary
+incompatibilities between APIs that chose different sides of the API
+tradeoff.
+
+In today's Rust, a library or program that chooses to use `u32`
+in "laughably small" cases would be incompatible with libraries
+that used the more conservative `u64`. This will result in more
+casting overall, likely making people less concerned when they
+perform more dangerous, truncating casts.
+
+We have previously discussed the possibility of automatically
+performing "widening" casts (`u32` to `u64`). This would allow
+libraries to freely be as conservative as they want, and
+reserve casts for dangerous situations that truly require
+thought.
+
+# Alternatives
+
+We could decide not to issue guidance here, instead relying
+on sub-ecosystems in Rust to make decisions appropriate to
+their domains. The drawback of that approach is that developers
+new to Rust will lack the benefit of any heuristic, and will
+be more likely to cargo-cult solutions.
+
+We could decide to encourage the use of `u64` even for small
+integers. While this would have the benefit of being
+significantly simpler, it would also mean that programs using
+tiny numbers would be unnecessarily slow. Because of this,
+it is doubtful that many people would even follow such a
+guideline, making it a dead letter and being practically
+similar to having no guideline at all.
+
+We could also encourage the use of `u32` as a default integer,
+discouraging the use of `u64` unless the user has a specific
+reason to believe the number would exceed `u32`. This has
+two problems.
+
+1. This guideline doesn't work very well in libraries,
+   which don't usually have a very concrete understanding
+   of what integers they will be used with.
+2. We want to discourage the use of `u32` in situations
+   where 32-bit overflow is a possibility. The
+   "laughably smaller" guideline, originally proposed by
+   @Valloric, captures most of the performance-critical
+   cases involving small numbers without reintroducing
+   significant opportunities for 32-bit overflows.
+
+# Unresolved questions
+
```

Comment:
It wasn't intended to be relevant or productive. I'm not here to help Mozilla increase their productivity by greasing the wheels and keeping threads on track. I respect people as a default but it can be lost, as is the case here. The example set by the core developers is that everything is personal. They're allowed to do whatever they want and anyone who speaks up against them is silenced by appealing to their own authority and policies. Rust's code of conduct isn't worth anything because it's just another policy to selectively apply and abuse.